File: binary_search.js
function binarySearch(sortedArray, num) {
    var found = false;
    var beginning = 0;
    var end = (sortedArray.length) - 1;
    while(beginning <= end && found === false){
        var middle = Math.floor((beginning + end) / 2);
        if(sortedArray[middle] === num) {
            found = true;
        } else if(sortedArray[middle] < num) {
            beginning = middle + 1
        } else {
            end = middle - 1
        }
    }
    return found;
}

function binarySearchIndex(sortedArray, num){
    var index = null;
    var beginning = 0;
    var end = (sortedArray.length) - 1;
    while(beginning <= end){
        var middle = Math.floor((beginning + end) / 2);
        if(sortedArray[middle] === num) {
            index = middle;
            break;
        } else if(sortedArray[middle] < num) {
            beginning = middle + 1;
        } else {
            end = middle - 1;
        }
    }
    return index;
}

var sortedArray = [1,3,7,9,13,15,24,56,89,99];
console.log(binarySearchIndex(sortedArray, 99));

File: binary_search.py
def binary_search(sorted_list, num):
    """Returns true or false depending on if num is in list"""

    found = False
    beginning = 0
    end = len(sorted_list) - 1

    while (beginning <= end) and not found:
        middle = (beginning + end) / 2
        if sorted_list[middle] == num:
            found = True
        elif sorted_list[middle] < num:
            beginning = middle + 1
        else:
            end = middle - 1

    return found

def binary_search_index(sorted_list, num):
    
    beginning = 0
    end = len(sorted_list) - 1

    while beginning <= end:
        middle = (beginning + end) / 2
        if sorted_list[middle] == num:
            return middle
        elif sorted_list[middle] < num:
            beginning = middle + 1
        else:
            end = middle - 1

sorted_list = [1,3,6,8,9,12,16,20]
print binary_search(sorted_list, 22) # False
print binary_search(sorted_list, 9) # True

print binary_search_index(sorted_list, 9) # 4
print binary_search_index(sorted_list, 2) # None